sky/spot/recovery_strategy.py

```diff
@@ -161,87 +221,43 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
-                    continue
-                if cluster_status != global_user_state.ClusterStatus.UP:
-                    # The cluster can be preempted before the job is launched.
-                    # Break to let the retry launch kick in.
-                    logger.info('The cluster is preempted before the job '
-                                'is submitted.')
-                    # TODO(zhwu): we should recover the preemption with the
-                    # recovery strategy instead of the current while loop.
-                    retry_launch = True
-                    break
-
-                try:
-                    status = spot_utils.get_job_status(self.backend,
-                                                       self.cluster_name)
-                except Exception as e:  # pylint: disable=broad-except
-                    # If any unexpected error happens, retry the job checking
-                    # loop.
-                    # Note: the CommandError is already handled in the
-                    # get_job_status, so it should not happen here.
-                    # TODO(zhwu): log the unexpected error to usage collection
-                    # for future debugging.
-                    logger.info(f'Unexpected exception: {e}\nFailed to get the '
-                                'job status. Retrying.')
-                    continue
-
-                # Check the job status until it is not in initialized status
-                if status is not None and status > job_lib.JobStatus.INIT:
-                    try:
-                        job_submitted_at = spot_utils.get_job_timestamp(
-                            self.backend, self.cluster_name, get_end_time=False)
-                        return job_submitted_at
-                    except Exception as e:  # pylint: disable=broad-except
-                        # If we failed to get the job timestamp, we will retry
-                        # job checking loop.
-                        logger.info(f'Unexpected Exception: {e}\nFailed to get '
-                                    'the job start timestamp. Retrying.')
-                        continue
-                # Wait for the job to be started
-                time.sleep(spot_utils.JOB_STARTED_STATUS_CHECK_GAP_SECONDS)
-
-            assert retry_launch
+            else:
+                # At this point, a sky.launch() has succeeded. Cluster may be
+                # UP (no preemption since) or DOWN (newly preempted).
+                job_submitted_at = self._wait_until_job_starts_on_cluster()
+                if job_submitted_at is not None:
+                    return job_submitted_at
+                # The job fails to start on the spot cluster, retry the launch.
+                # TODO(zhwu): log the unexpected error to usage collection
+                # for future debugging.
+                logger.info('Failed to get the job status, due to unexpected '
+                            'job submission error.')
```

Comment:
```suggestion
                logger.info('Failed to successfully submit the job to the launched cluster, due to unexpected submission errors or the cluster being preempted during job submission.')
```
(This could be expected if the cluster is preempted right after launching?)